flytekit/extras/sqlite3/task.py

```diff
@@ -84,16 +92,76 @@ def output_columns(self) -> typing.Optional[typing.List[str]]:
         c = self.python_interface.outputs["results"].column_names()
         return c if c else None
 
-    def execute(self, **kwargs) -> typing.Any:
+    def get_custom(self, settings: SerializationSettings) -> typing.Dict[str, typing.Any]:
+        return {
+            "query_template": self.query_template,
+            "inputs": "",
+            "outputs": "",
+            "uri": self.task_config.uri,
+            "compressed": self.task_config.compressed,
+        }
+
+    def get_command(self, settings: SerializationSettings) -> typing.List[str]:
+        return [
+            "pyflyte-manual-execute",
+            "--inputs",
+            "{{.input}}",
+            "--output-prefix",
+            "{{.outputPrefix}}",
+            "--raw-output-data-prefix",
+            "{{.rawOutputDataPrefix}}",
+            "--task_executor",
+            f"{SQLite3TaskExecutor.__module__}.{SQLite3TaskExecutor.__name__}",
+            "--task_template_path",
+            "{{.taskTemplatePath}}",
+        ]
+
+    def serialize_to_template(self, settings: SerializationSettings) -> task_models.TaskTemplate:
+        # This doesn't get called from translator unfortunately. Will need to move the translator to use the model
+        # objects directly first.
+        obj = task_models.TaskTemplate(
+            identifier_models.Identifier(
+                identifier_models.ResourceType.TASK, settings.project, settings.domain, self.name, settings.version
+            ),
+            self.task_type,
+            self.metadata.to_taskmetadata_model(),
+            self.interface,
+            self.get_custom(settings),
+            container=self.get_container(settings),
+            config=self.get_config(settings),
+        )
+        self._task_template = obj
+        return obj
+
+
+class SQLite3TaskExecutor(TaskTemplateExecutor[SQLite3Task]):
+
+    @classmethod
+    def promote_from_template(cls, tt: task_models.TaskTemplate) -> SQLite3Task:
+        qt = tt.custom["query_template"]
+
+        return SQLite3Task(
```

Comment:
why do we need this? What this reads as is that TaskTemplateExecutor will always need to be able to create a task from the template?

How about a simpler interface
TaskTemplateExecutor().execute(TaskTemplate, inputs)?